src/Utils/Crypto.php

```diff
@@ -41,11 +42,59 @@ public static function decrypt( $payload, $cipherParams ) {
 
     /**
      * Returns default encryption parameters.
-     * @param $key string|null Encryption key, if not provided a random key is generated.
+     * @param $params Array Array containing optional cipher parameters. A `key` must be specified.
+     * The key may be either a binary string or a base64 encoded string, in which case `'base64Key' => true` must be set.
+     * `iv` can also be provided as binary or base64 string (`'base64IV' => true`), although you shouldn't need it in most cases.
      * @return CipherParams Default encryption parameters.
      */
-    public static function getDefaultParams( $key = null ) {
-        return new CipherParams( $key );
+    public static function getDefaultParams( $params ) {
+        if ( !isset( $params['key'] ) ) throw new AblyException ( 'No key specified.', 40003, 400 );
+
+        $cipherParams = new CipherParams();
+
+        if ( isset( $params['base64Key'] ) && $params['base64Key'] ) {
+            $params['key'] = strtr( $params['key'], '_-', '/+' );
+            $params['key'] = base64_decode( $params['key'] );
+        }
+
+        $cipherParams->key = $params['key'];
+        $cipherParams->keyLength = isset( $params['keyLength'] ) ? $params['keyLength'] : strlen( $cipherParams->key ) * 8;
+        
+        if ( !in_array( $cipherParams->keyLength, array( 128, 256 ) ) ) {
```

Comment:
This only applies if the algorithm is `aes`, if not, we don't enforce this even though we're not sure what other algorithms may or may not work